### ISSUE TITLE
Allel finding continues finding more and more allels

### ISSUE THREAD

**Issue by krdav** [psathyrella/partis#210]
I tried the new allel finding feature. The concept is really cool and better still; very important, so props for taking it up. My experience running it was not that good though. I ran it on a bunch of sequences from a single animal with a pretty well defined germline gene set where all the germlines should be included in the IMGT set and therefore already be in partis. Therefore I expected that the allel finding feature would only find a very limited set of new allels that I might have missed. 

Instead it found dusins and restarted finding new allels just to find more. In conclusion it looked like overfitting on the shm's.

Can you elaborate a little on how this algorithm works and what mechanisms there are in place to avoid overfitting on the shm's? Also if you would like to test it yourself, let me know then I will prepare a dataset for you.

**Comment by psathyrella** [comment]
yeah, sorry, writing up a good description of what it's doing has been lurking on my todo list for too long, I'll do that today.

It's supposed to keep running repeatedly until it doesn't find any new alleles -- this is how it deals with repertoires that have more than one new allele for a single gene. But two things for your case
  - it's definitely still in beta, and the thing that's limiting how well I can validate it is figuring out how to cover a reasonable amount of the space of possible ways that repertoires can make it fail, so yes! if you could send the data that would be great.
  -  one of the biggest differences between the bleeding edge version (germline set generation is mostly what I've been working on in dev) and what's in master is I've made it deal a lot better with low sample sizes assigned to a gene, which is my first guess as to why it'd turn up a bunch of spurious alleles.

**Comment by krdav** [comment]
Don't push it for me, I don't need it right now. You got the data and their associated germlines.
